File: Programmers/위장.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> hashMap = new HashMap<>();

        for (int i = 0; i < clothes.length; i++) {
            hashMap.put(clothes[i][1], hashMap.getOrDefault(clothes[i][1], 0) + 1);
        }

        for (int value : hashMap.values()) {
            answer *= (value + 1);
        }

        return answer - 1;
    }
}

File: Programmers_python/위장.py
from collections import defaultdict

def solution(clothes):
    answer = 1
    clothes_dict = defaultdict(list)

    for c in clothes:
        clothes_dict[c[1]].append(c[0])

    for key, value in clothes_dict.items():
        answer *= len(value)+1

    return answer - 1
